lib/apress/yandex_market/readers/model_by_category.rb

```
@@ -0,0 +1,145 @@
+      #          :vendor=>{:name=>"Caviale"},
+      #          :specification=>[
+      #            {
+      #              :features=>[
+      #                {:value=>"время нанесения: дневной/ночной"},
+      #                {:value=>"эффект: увлажнение"}
+      #              ]
+      #            }
+      #          ]
+      #        }
+      #        {
+      #          :__line__=>2,
+      #          :__column__=>1,
+      #          :id=>217189256,
+      #          ...
+      #        }
+      class ModelByCategory < Base
+        FIELDS = %w(
+          MODEL_CATEGORY
+          MODEL_PHOTO
+          MODEL_PHOTOS
+          MODEL_PRICE
+          MODEL_SPECIFICATION
+          MODEL_VENDOR
+        ).join(',').freeze
+
+        class << self
+          def allowed_options
+            Readers::Category.allowed_options
+          end
+        end
+
+        attr_reader :category_reader
+
+        # Public: инициализация ридера
+        #
+        # options - Hash, параметры ридера
+        #   :region_id  - идентификатор региона (необязательный)
+        #   :token      - токен для доступа к API Яндекс.Маркета
+        #   :categories - список категорий для загрузки моделей Яндекс.Маркета
+        #
+        # Returns an instance of Readers::ModelByCategory
+        def initialize(options)
+          super
+          @category_reader = Readers::Category.new(options)
+        end
+
+        # Public: читаем товары из API и фильтруем необходимые поля для использования далее в препроцессоре Кирби
+        #
+        # Returns nothing
+        def each_row
+          presenter = Presenters::Model.new
+
+          category_reader.each_row do |category|
+            models = Set.new
+
+            begin
+              process_models(models, category.fetch(:id), 'DESC')
+            rescue Api::PageError
+              begin
+                process_models(models, category.fetch(:id), 'ASC')
+              rescue Api::PageError
```

Do not suppress exceptions.